static/js/common/DropdownMenu.jsx

```diff
@@ -0,0 +1,55 @@
+import React, {useEffect, useRef, useState} from 'react';
+import PropTypes from "prop-types";
+
+
+const DropdownMenu = ({children, buttonContent}) => {
+    /**
+     * buttonContent is a React component for the opening/closing of a button.
+     * the menu will be closed in click anywhere except in an element with classname 'preventClosing'.
+     * this class is using useRef for open/close rather than useState, for changing state triggers re-rendering of the
+     * component and all its children, so when clicking on children their onClick won't be executed.
+     */
+
+    const dropdownLinksClass = 'dropdownLinks-menu';
+    const isOpenRef = useRef(null);
```

Comment:
this is actually referring to the dropdown elm. Considure a name like "dropdownRef". In line 26 you can have a function `isOpen(dropdownRef)` which returns `dropdownRef.current?.className?.includes('open')` for something that is very readable